src/ansys/fluent/core/filereader/casereader.py

```diff
@@ -170,3 +175,35 @@ def _get_processed_string(input_string: bytes) -> str:
     rp_vars_str = codecs.decode(input_string, errors="ignore")
     string_identifier = "(37 ("
     return string_identifier + rp_vars_str.split(string_identifier)[1]
+
+
+def _get_case_filepath(project_dir_path: str) -> str:
+    """Gets path of any file having extension either of .cas, .cas.gz or.
+
+    .cas.h5 and returns the same.
+
+    Parameters
+    ----------
+    project_dir_path : str
+        The directory containing the case file
+
+    Returns
+    -------
+    case file path (str)
+    """
+    file_list = list(
+        itertools.chain(
+            *(
+                glob.glob(project_dir_path + r"\**\**-Solve\*.%s" % ext)
```

Comment:
I think the case-filepath information should be extracted from the .flprj file. @lagacep-ans Do we have a document on .flprj layout?